src/openrct2-ui/windows/Park.cpp

```diff
@@ -1099,7 +1099,12 @@ class ParkWindow final : public Window
         }
         else
         {
-            ft.Add<uint16_t>(gameState.ScenarioObjective.NumGuests);
+            if (gameState.ScenarioObjective.Type == OBJECTIVE_GUESTS_BY
+                || gameState.ScenarioObjective.Type == OBJECTIVE_GUESTS_AND_RATING)
+                ft.Add<int32_t>(gameState.ScenarioObjective.NumGuests);
+            else
+                ft.Add<uint16_t>(gameState.ScenarioObjective.NumGuests);
```

Comment:
What? are we using this field for multiple different things?

Reply:
Ahh i see what is happening. I don't like this there is a high chance that it will go wrong.

Reply:
As far as I can tell from the usages no, but this code is used to print multiple messages, i.e the next string which uses the same code pops 16 bytes off,
```
STR_2392    :{BLACK}To achieve a monthly income from ride tickets of at least {POP16}{POP16}{CURRENCY}
```

Reply:
You see how OBJECTIVE_BUILD_THE_BEST is separated out just do the same for the objectives with guest counts